multicluster/controllers/multicluster/resourceexport_controller.go

```diff
@@ -47,16 +71,302 @@ type ResourceExportReconciler struct {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.8.3/pkg/reconcile
 func (r *ResourceExportReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	klog.InfoS("reconciling ResourceExport", "resourceexport", req.NamespacedName)
+	var resExport mcsv1alpha1.ResourceExport
+	if err := r.Client.Get(ctx, req.NamespacedName, &resExport); err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	resImportName := types.NamespacedName{
+		Name:      getResourceImportName(&resExport),
+		Namespace: req.Namespace,
+	}
+
+	if !resExport.DeletionTimestamp.IsZero() {
+		if common.ContainsString(resExport.Finalizers, common.ResourceExportFinalizer) {
```

Comment:
I am not familiar with the Finalizer concept. Could you give some explanation or share me some documents about Finalizer?

Reply:
sure, the finalizers are actually a string array which can prevent object being deleted directly. https://book.kubebuilder.io/reference/using-finalizers.html
https://kubernetes.io/docs/concepts/overview/working-with-objects/finalizers

Reply:
Thanks!